chia/rpc/wallet_rpc_api.py

```diff
@@ -1500,9 +1507,18 @@ async def nft_get_nfts(self, request) -> EndpointResult:
         wallet_id = uint32(request["wallet_id"])
         nft_wallet: NFTWallet = self.service.wallet_state_manager.wallets[wallet_id]
         nfts = nft_wallet.get_current_nfts()
+        start_index = request.get("start_index", 0)
+        num = request.get("num", len(nfts))
         nft_info_list = []
+        count = 0
         for nft in nfts:
-            nft_info_list.append(nft_puzzles.get_nft_info_from_puzzle(nft))
+            if count >= start_index and count - start_index < num:
+                nft_info_list.append(
+                    nft_puzzles.get_nft_info_from_puzzle(
+                        nft, self.service.wallet_state_manager.config.get(CACHE_PATH_KEY, None)
```

Comment:
I think including/fetching the off-chain metadata should be optional. Can we use a request param `"include_off_chain_metadata"` to control that?